script.js

```diff
@@ -1,4 +1,181 @@
+//Roll the dice and put them in order to make the highest number possible. If you roll a 4 and an 6, for example, your best answer would be 64. Using 3 dice, a roll of 3, 5 and 2 should give you 532, and so on. Write down your answer, pass the dice, and challenge the next player to Beat That!
+
+////Game State for stages of the gameplay
+var gameState = "";
+
+//Global variable for number of players and initialize player 1
+var numberOfPlayers = 0;
+var playerNumber = 1;
+
+//Game state for highest or lowest combined score
+var gameMode = "";
+
+//Global List for recording player rolls
+var playersList = [];
+
+//Global Player Number sum & score
+var playerScore = [];
+
+var leaderBoard = [];
+
+//Numnber of dice to roll for game
+var numberOfDice = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "") {
+    //User to choose game mode
+    //Check for valid input
+    if (input != "highest" && input != "lowest") {
+      return `Please choose highest or lowest number mode.`;
+    } else {
+      gameMode = input.toLowerCase();
+      //If number of players is set, move to game
+      if (numberOfPlayers != 0) {
+        gameState = "game";
+      }
+      //Else move to number of players
+      else {
+        gameState = "playersSelection";
+      }
+      return `You have selected ${input} number mode. Please enter number of players`;
+    }
+  }
+
+  //User to select number of players
+  if (gameState == "playersSelection") {
+    //Check for valid input
+    if (isNaN(input) || input == "") {
+      return `Error! Please enter number of players`;
+    } else {
+      numberOfPlayers = input;
+      //Initialize array for number of players
+      for (i = 0; i < numberOfPlayers; i++) {
+        playerScore[i] = 0;
+      }
+      //Move to dice selection
+      gameState = "diceSelection";
+      return `There will be ${numberOfPlayers} players in the game. Please enter number of dices.`;
+    }
+  }
+  //User enter how many dices to play
+  if (gameState == "diceSelection") {
+    numberOfDice = Number(input);
+    //Check input for valid number
+    if (isNaN(numberOfDice) || input == "") {
+      return `Not a number! Please select number of dices to roll`;
+    }
+    //Move to next game mode
+    gameState = "game";
+    return `All players will now be playing with ${numberOfDice} dices`;
+  }
+
+  //Game Function
+  if (gameState == "game") {
+    //Option for user to change game mode halfway
+    if (input == "change") {
+      gameMode = "";
+      return `Change Mode! Please choose highest or lowest`;
+    }
+
+    while (playerNumber <= numberOfPlayers) {
+      var normal = normalGame();
```

Comment:
You are calling playersSum here as a new array within the function

Generally in the context of the game, would be good to have a global variable up top and affect the global variable (easier to keep track) but this is preference. No right or wrong